credentials/apps/badges/admin.py

```diff
@@ -145,20 +147,89 @@ def image(self, obj):
     image.short_description = _("icon")
 
 
+class DataRulePenaltyInline(admin.TabularInline):
+    model = PenaltyDataRule
+    extra = 0
+    fields = [
+        "data_path",
+        "operator",
+        "value",
+    ]
+
+    def get_readonly_fields(self, request, obj=None):
+        readonly_fields = list(super().get_readonly_fields(request, obj))
+        if obj and obj.requirement.template.is_active:
+            readonly_fields.extend([
+                "data_path",
+                "operator",
+                "value",
+            ])
+        return readonly_fields
+
+
+class BadgePenaltyAdmin(admin.ModelAdmin):
+    """
+    Badge requirement penalty setup admin.
+    """
+    inlines = [
+        DataRulePenaltyInline,
+    ]
+
+    list_display = [
+        "id",
+        "requirement",
+    ]
+    list_display_links = (
+        "id",
+        "requirement",
+    )
+    list_filter = [
+        "requirement",
+        "description",
+    ]
+
+
+    def get_readonly_fields(self, request, obj=None):
+        readonly_fields = list(super().get_readonly_fields(request, obj))
+        if obj and obj.requirement.template.is_active:
+            readonly_fields.extend([
+                "requirement",
+            ])
+        return readonly_fields
+
+
+class BadgePenaltyInline(admin.TabularInline):
+    model = BadgePenalty
+    extra = 0
+    fields = [
+        "effect",
+        "description",
+    ]
+
+    def get_readonly_fields(self, request, obj=None):
+        readonly_fields = list(super().get_readonly_fields(request, obj))
+        if obj and obj.template.is_active:
+            readonly_fields.extend([
+                "effect",
+                "description",
+            ])
+        return readonly_fields
+
+
 class BadgeRequirementAdmin(admin.ModelAdmin):
     """
     Badge template requirement admin setup.
     """
 
     inlines = [
         DataRuleInline,
+        BadgePenaltyInline,
```

Comment:
Move to a badge template instead:

- BadgeTemplate
  - Requirements
    - data rules
  - Penalties 
    - data rules

Reply:
1 Penalty can affect more than 1 Requirement progress.